Program.cs

```diff
@@ -1,4 +1,78 @@
+// ------------------------------------------------------------------------
+// Program.cs
+// SORT AND SWAP SPECIAL CHARACTERS 
+// Given a character array A, along with special character S and sort order O (default order is Ascending)
+// Print the sorted array by keeping the elements matching S to the last of the array.
+// For example, Input=([a, b, c, a, c, b, d], a, “descending”) ;Output=d, c, c, b, b, a, a .
+// --------------------------------------------------------------------------------------------
+using System;
+using System.Linq;
+namespace Training {
+   #region Program ------------------------------------------------------------------------------
+   /// <summary>Sort and swap special character</summary>
+   internal class Program {
+      #region Methods ---------------------------------------------
+      /// <summary>Getting inputs from user</summary>
+      static void Main () {
+         // Getting character from the user.
+         char[] a = GetInput ("Enter the letters: ").ToCharArray ();
+         // Getting special character to swap from the user.
+         char s = char.Parse (GetInput ("Enter the special character: "));
+         // Getting the sorting order from the user. 
+         while (true) {
+            Console.Write ("Do you want to sort the character array in decending order: (y)es or (n)o: ");
+            ConsoleKeyInfo consoleKey = Console.ReadKey ();
+            ConsoleKey sortOrder = consoleKey.Key;
+            if (sortOrder == ConsoleKey.Y) {
+               Console.Write ($"\nOutput: {SortWithSpecialChar (a, s, "descending")}");
+               break;
+            } else if (sortOrder == ConsoleKey.N) {
+               Console.Write ($"\nOutput: {SortWithSpecialChar (a, s)}");
+               break;
+            } else Console.WriteLine ("\nInvalid key. Please enter y or n.");
+         }
+      }
 
-   
\ No newline at end of file
+      /// <summary>Provide valid input</summary>
+      /// <param name="prompt">Comment for the user</param>
+      /// <returns>Valid input</returns>
+      static string GetInput (string prompt) {
+         string result;
+         while (true) {
+            Console.Write (prompt);
+            string input = Console.ReadLine ();
+            if (!string.IsNullOrWhiteSpace (input) && input.All (char.IsLetter)) {
+               result = input;
+               break;
+            } else Console.WriteLine ("Invalid input. Please enter alphabets only and make sure it's not empty.");
+         }
+         return result;
+      }
+
+      /// <summary>Sort the character array with special character</summary>
+      /// <param name="a">Character array</param>
+      /// <param name="s">Special character</param>
+      /// <param name="order">Order for sorting the character</param>
+      /// <returns>Sorted character</returns>
+      static string SortWithSpecialChar (char[] a, char s, string order = "ascending") {

```

Comment:
Instead of having `order` as a string use a `bool` instead. That way, the string check will be eliminated